### ISSUE TITLE
extra-settings change possible?

### ISSUE THREAD

**Issue by dsh21** [dotansimha/angularjs-dropdown-multiselect#76]
.jsp

<div id="mddPod" ng-dropdown-multiselect="" search-filter="mddPodFilter"
		options="mddPodData" selected-model="mddPodModel"
		extra-settings="mddPodSettings" events="mddPodEvents" checkboxes="true"></div>

.js

$scope.mddPodModel 		= [];
$scope.mddPodData 		= [];
$scope.mddPodFilter 	= "";
$scope.mddPodSettings 	= {	enableSearch 	: false,
							displayProp 	: "NAME",
							idProp 			: "CODE",
							externalIdProp  : "CODE",
							showCheckAll	: false,
							showUncheckAll  : false,
							scrollable		: true,
							scrollableHeight: "150px",
							smartButtonMaxItems: 0,
							selectionLimit: 0
};

i want to change another "extra-settings". is it possible?? this don't change....

if($scope.mddPodData.length > 15){
	
		$scope.mddPodSettings 	= {	
				scrollable		: true,
				scrollableHeight: "300px"
		};
	
}else{
	
		$scope.mddPodSettings 	= {	
				scrollable		: false,
		};
	
}

**Comment by shonatodc** [comment]
Yeah doesn't seem that we can pass models to that attribute value for some reason...

**Comment by RamakriJaya** [comment]
I am also facing issues with the extra-settings on my page
Though I initialise extra-settings directive, it does not resolve to appropriate values after the compile phase
It always displays the id of the object being rendered but whereas I want to display the label.

Thanks

**Comment by danielriosb** [comment]
any solution for this?

**Comment by pkempenaers** [comment]
What do you mean with "this", there are multiple "issues" being discussed

**Comment by sudarsangp** [comment]
@dsh21 You can assign the changes to a variable and then finally assign it to `$scope`.

    var mddPodSettings
    if ($scope.mddPodData.length > 15) {
        mddPodSettings   = { 
            scrollable      : true,
            scrollableHeight: "300px"
        };
    } else {
        mddPodSettings   = { 
            scrollable      : false,
        };
    }
    $scope.mddPodSettings = mddPodSettings;
